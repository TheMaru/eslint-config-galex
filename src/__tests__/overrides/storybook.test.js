const { createStorybookOverride } = require('../../overrides/storybook');

describe('createStorybookOverride', () => {
  test('matches snapshot if storybook is false', () => {
    const project = {
      storybook: {
        hasStorybook: false,
      },
      react: {
        hasReact: true,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    expect(createStorybookOverride(project)).toMatchSnapshot();
  });

  test('matches snapshot if storybook is present', () => {
    const project = {
      storybook: {
        hasStorybook: true,
      },
      react: {
        hasReact: true,
      },
      typescript: {
        hasTypeScript: true,
      },
    };

    expect(createStorybookOverride(project)).toMatchSnapshot();
  });

  test('matches snapshot if storybook is present, without react and ts', () => {
    const project = {
      storybook: {
        hasStorybook: true,
      },
      react: {
        hasReact: false,
      },
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createStorybookOverride(project)).toMatchSnapshot();
  });

  test('matches snapshot if storybook is absent, but storybook/testing-library is present', () => {
    const project = {
      storybook: {
        hasStorybook: false,
        hasStorybookTestingLibrary: true,
      },
      react: {
        hasReact: false,
      },
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createStorybookOverride(project)).toMatchSnapshot();
  });

  test('matches snapshot if both storybook and but storybook/testing-library are present', () => {
    const project = {
      storybook: {
        hasStorybook: true,
        hasStorybookTestingLibrary: true,
      },
      react: {
        hasReact: false,
      },
      typescript: {
        hasTypeScript: false,
      },
    };

    expect(createStorybookOverride(project)).toMatchSnapshot();
  });
});
