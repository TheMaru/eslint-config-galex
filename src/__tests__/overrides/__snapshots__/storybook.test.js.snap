// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStorybookOverride matches snapshot if both storybook and but storybook/testing-library are present 1`] = `
{
  "files": [
    "*.stories.*",
    "*.story.*",
  ],
  "overrideType": "eslint-config-galex/storybook",
  "plugins": [
    "storybook",
  ],
  "rules": {
    "import/no-anonymous-default-export": "off",
    "import/no-default-export": "off",
    "storybook/await-interactions": "error",
    "storybook/context-in-play-function": "error",
    "storybook/csf-component": "error",
    "storybook/default-exports": "off",
    "storybook/hierarchy-separator": "warn",
    "storybook/meta-inline-properties": "warn",
    "storybook/no-redundant-story-name": "warn",
    "storybook/no-stories-of": "error",
    "storybook/no-title-property-in-meta": "warn",
    "storybook/prefer-pascal-case": "warn",
    "storybook/story-exports": "warn",
    "storybook/use-storybook-expect": "error",
    "storybook/use-storybook-testing-library": "error",
  },
}
`;

exports[`createStorybookOverride matches snapshot if storybook is absent, but storybook/testing-library is present 1`] = `null`;

exports[`createStorybookOverride matches snapshot if storybook is false 1`] = `null`;

exports[`createStorybookOverride matches snapshot if storybook is present 1`] = `
{
  "files": [
    "*.stories.*",
    "*.story.*",
  ],
  "overrideType": "eslint-config-galex/storybook",
  "plugins": [
    "storybook",
  ],
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": "off",
    "@typescript-eslint/no-empty-function": "off",
    "import/no-anonymous-default-export": "off",
    "import/no-default-export": "off",
    "react/function-component-definition": "off",
    "storybook/csf-component": "error",
    "storybook/default-exports": "off",
    "storybook/hierarchy-separator": "warn",
    "storybook/meta-inline-properties": "warn",
    "storybook/no-redundant-story-name": "warn",
    "storybook/no-stories-of": "error",
    "storybook/no-title-property-in-meta": "warn",
    "storybook/prefer-pascal-case": "warn",
    "storybook/story-exports": "warn",
    "storybook/use-storybook-expect": "error",
    "storybook/use-storybook-testing-library": "error",
  },
}
`;

exports[`createStorybookOverride matches snapshot if storybook is present, without react and ts 1`] = `
{
  "files": [
    "*.stories.*",
    "*.story.*",
  ],
  "overrideType": "eslint-config-galex/storybook",
  "plugins": [
    "storybook",
  ],
  "rules": {
    "import/no-anonymous-default-export": "off",
    "import/no-default-export": "off",
    "storybook/csf-component": "error",
    "storybook/default-exports": "off",
    "storybook/hierarchy-separator": "warn",
    "storybook/meta-inline-properties": "warn",
    "storybook/no-redundant-story-name": "warn",
    "storybook/no-stories-of": "error",
    "storybook/no-title-property-in-meta": "warn",
    "storybook/prefer-pascal-case": "warn",
    "storybook/story-exports": "warn",
    "storybook/use-storybook-expect": "error",
    "storybook/use-storybook-testing-library": "error",
  },
}
`;
